src/deluge/model/song/song.cpp

```diff
@@ -148,7 +148,7 @@ Song::Song() : backedUpParamManagers(sizeof(BackedUpParamManager)) {
 
 	swingAmount = 0;
 
-	swingInterval = 8 - insideWorldTickMagnitude; // 16th notes
+	swingInterval = FlashStorage::defaultSwingInterval;
```

Comment:
does this work if you change song resolution?

Reply:
Kinda no, kinda yes.

BUG: Defaults shows the swing interval in relationship to current song magnitude, not the default magnitude. If you shink default magnitude by one the default swing interval will look the same until you actually have a new song with the new magnitude, at which point it will have doubled -- and that will be the new song's initial swing interval. (I have a fix so that default swing interval is based on the default magnitude, so defaults shows the right value.) 

FEATURE OR BUG? With the above bug fixed, changing default magnitude still changes the default swing interval. We could mostly adjust the swing interval automatically when changing magnitude, but if you start with an extreme swing interval and change magnitude in the wrong direction, then we can't. (I don't have this done, but I can do it.)